apps/api/src/rating/rating.service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CreateRatingDto } from './dto/create-rating.dto';
+
+import { JwtService } from '@nestjs/jwt';
+import { AuthService } from 'src/auth/auth.service';
+import { Rating } from './schemas/rating.schema';
+
+import { extractIdFromToken } from 'src/utils/extract-token.utils';
+import { OrderService } from 'src/order/order.service';
+
+@Injectable()
+export class RatingService {
+  constructor(
+    @InjectModel(Rating.name) private readonly ratingModel: Model<Rating>,
+    private jwtService: JwtService,
+    private authService: AuthService,
+    private orderService: OrderService,
+  ) {}
+
+  async getUserRatings(req: any): Promise<Rating[]> {
+    const userId = await extractIdFromToken(req, this.jwtService);
+
+    return this.ratingModel.find({ userId });
+  }
+
+  async createOrderRating(
+    req: any,
+    createRatingDto: CreateRatingDto,
+  ): Promise<Rating> {
+    const userId = await extractIdFromToken(req, this.jwtService);
+
```

Comment:
it might be more efficient to utilize the userId from the request payload
```suggestion
    const userId = req._id;
```